### DISCUSSION TITLE
Client download from remote URL in SvelteKit?

### DISCUSSION THREAD

**Discussion by Vervo7** [sveltejs/kit#3893] [Q&A] [answered]
In PHP, I just do something like this:

$file_url = 'https://www.someremoteserver.com/remotefile.pdf';
header('Content-Type: application/octet-stream');
header('Content-Transfer-Encoding: Binary'); 
header('Content-disposition: attachment; filename=custom-file-name.pdf' ); 
readfile($file_url);

This forces a client download without exposing the original URL.

How can I do this in SvelteKit?

**Comment by mustofa-id** [comment, marked as answer]
I believe you can do it using `fetch`.

```js
export const get = async () => {
	const res = await fetch(`your url`)
	return {
		headers: { /* your headers */ },
		body: res.body
	}
}
```

**Reply by Vervo7** [reply]
Thanks, mustofa-id! I was using https://www.npmjs.com/package/got but fetch seems a better solution.

**Reply by mustofa-id** [reply]
sure, so you don't need additional dependency.
